app/component/SummaryPage.js

```diff
@@ -1519,7 +1521,13 @@ class SummaryPage extends React.Component {
           alternativePlan: undefined,
         },
         () => {
-          if (relevantRoutingSettingsChanged(this.context.config)) {
+          const hasNonWalkingItinerary = this.selectedPlan?.itineraries?.some(
+            itinerary => !itinerary.legs.every(leg => leg.mode === 'WALK'),
+          );
+          if (
+            relevantRoutingSettingsChanged(this.context.config) &&
+            hasNonWalkingItinerary
```

Comment:
Is this really what's intended here? If we *guess* what's intended, shouldn't we check the user-configured modes (`getSettings().modes`) if there's *only* walking enabled? Is it really the right approach to inspect the returned itineraries?

We stumbled upon this when merging from HSL's repo into our bbnavi fork, and have trouble understanding why this change has been made, despite the PR's description. 😕 – Maybe we're missing some context here.

We think a good way forward would be to document such intricate logic – which effectively make up significant parts of digitransit-ui's (intended and unintended) behaviour – in a more permanent way, e.g. in `docs/Routing.md`.

cc @vesameskanen